### ISSUE TITLE
Le livre au format ebook

### ISSUE THREAD

**Issue by smimram** [LaurentClaessens/mazhe#13]
Bonjour,

Ça serait super si le livre était aussi disponible au format ebook (ou mobi, etc) pour être consultable sur une liseuse. Il semble exister des outils pour faire cela, par exemple [tex4ebook](https://github.com/michal-h21/tex4ebook), mais je n'ai pas encore eu l'occasion de les tester.

Merci !

**Comment by LaurentClaessens** [comment]
Le 29/09/2016 à 16:16, Samuel Mimram a écrit :
> Bonjour,
>
> Ça serait super si le livre était aussi disponible au format ebook
> (ou mobi, etc) pour être consultable sur une liseuse. Il semble
> exister des outils >pour faire cela, par exemple
> [tex4ebook](https://github.com/michal-h21/tex4ebook), mais je n'ai
> pas encore eu l'occasion de les tester.


Je pensais (naïvement ?) que les liseuses pourraient lire les PDF.
Il doit y avoir moyen de faire pdf->epub parce que thebookedition.com 
propose de vendre au format epub alors que je ne leur fournis que le pdf.

**Comment by Naereen** [comment]
Oui, par exemple :
https://cloudconvert.com/pdf-to-epub

Reste a trouver une solution simple et libre sur Linux.

**Comment by Naereen** [comment]
Et https://cloudconvert.com/pdf-to-mobi aussi.

**Comment by LaurentClaessens** [comment]
Oui, il faudrait un truc simple et surtout en ligne de commande parce que je veux que les processus 
latex->pdf->mise à jour sur mon site 
latex->git commit->git push
soient automatisés.
Je n'ai rien contre ajouter une ligne pdf2epud à mes scripts, mais j'ai quelque chose contre le fait de passer par un service tiers qui me demande de faire des clics à chauqe mise à jour.

Si quelqu'un veut s'y coller, j'y vois pas d'inconvénients; je pourrais même travailler à mettre une alerte RSS automatique à chaque changement de pdf.

**Comment by smimram** [comment]
Merci pour vos réponses, voici quelques précisions.

La plupart des liseuses arrivent bien à lire du pdf. Le problème c'est que généralement l'écran d'une liseuse est bien plus petit qu'une page A4 et donc il est souhaitable de changer la mise en page (enlever les marges, avoir moins de mots par lignes, recentrer les formules, etc.). Ceci est malheureusement impossible avec un pdf (qui stocke grosso-modo les pages sous formes d'images + un peu de contenu texte), d'où l'intérêt des formats dédiés de type mobi ou ebook. Pour cette même raison, la conversion automatique depuis le format pdf est généralement de mauvaise qualité (ça marche pour du texte simple, genre un roman, mais pour les documents mathématiques avec des formules ça devient vite la foire) et la génération d'un ebook à partir des sources tex serait vraiment souhaitble.

J'aurais bien fait quelques essais mais je dois avouer que la complexité du système de compilation m'a un peu découragé... Je pourrais arriver à avoir quelque chose qui marche sur un petit cas de test si vous voulez.

**Comment by Naereen** [comment]
Je comprends mieux le problème. Une solution tex → epub / mobi serait idéale.

Essayez de faire un test sur un petit fichier LaTeX (avec des maths), et si ça fonctionne ça devrait pouvoir s'utiliser pour le Frido entier sans soucis.

**Comment by LaurentClaessens** [comment]
Chez moi, la commande
```
tex4ebook mazhe.tex
```
fait des choses.
Ensuite
```
tex4ht mazhe.tex
```
a l'air de reprendre le travail pour sortir du html.

Problème : il a l'air de ne pas aimer les math à l'intérieur de \text.
```
\begin{equation}
\text{ si \(a\) est nul }
\end{equation}
```

**Comment by Naereen** [comment]
Je crois que, sémantiquement, des maths ne devraient pas être à l'intérieur de `\text`, ce qui expliquerait le problème rencontré avec `tex4ebook`, `tex4ht`.

Un remplacement automatique à coup de *regexp* devrait pouvoir corriger ça.
[Par exemple](https://regex101.com/r/3sI7mC/1) :

```bash
# /!\ a tester d'abord
sed --in-place=tmp s/'\\text\{([^\(\)]*)\\\(([^\(\)]*)\\\)([^\(\)]*)\}'/'\\text\{\1\} \\\(\2\\\) \\text\{\3\}'/g */*.tex
```

**Comment by LaurentClaessens** [comment]
Les math dans \text{...} sont éliminées. J'espère qu'il n'y aura pas trop de problèmes d'allignement.

Pour `tex4ebook`, on n'y est cependant pas encore. Je crois qu'il passe par DVI et qu'il lui manque un certain nombre de choses de `pdflatex`.

* Il plante à l'ultilsation de l'environnement `problème` (voir `e_mazhe.tex`) avec l'erreur 
```
! LaTeX Error: There's no line here to end.
```

* Il plante aussi sur ceci :
```
(./pictures_tex/Fig_MCKyvdk.pstricks
! Missing number, treated as zero.
<to be read again> 
                   }
l.54 ...figuresmazheDOTpy}{\totalheightof{\( A\)}}
```
Mais ça c'est moins grave parce que c'est une ligne qui sert à écrire la taille de la boîte $A$ dans un fichier intermédiaire, en vue de regénérer la figure en en tenant compte. Dès que les fichiers PDF sont générés, on peut en réalité remplacer tous les `*.pstricks` par des    `\includegraphics`
